x-pack/plugins/lens/public/visualizations/metric/dimension_editor.test.tsx

```diff
@@ -5,64 +5,38 @@
  * 2.0.
  */
 
-/* eslint-disable max-classes-per-file */
-
-import React, { FormEvent } from 'react';
-import { OperationDescriptor, VisualizationDimensionEditorProps } from '../../types';
+import React from 'react';
+import { render, screen, waitFor as waitForRtl } from '@testing-library/react';
+import faker from 'faker';
+import userEvent from '@testing-library/user-event';
+import { chartPluginMock } from '@kbn/charts-plugin/public/mocks';
+import { euiLightVars } from '@kbn/ui-theme';
 import { CustomPaletteParams, PaletteOutput, PaletteRegistry } from '@kbn/coloring';
-
+import { VisualizationDimensionEditorProps } from '../../types';
 import { MetricVisualizationState } from './visualization';
 import {
   DimensionEditor,
   DimensionEditorAdditionalSection,
   SupportingVisType,
 } from './dimension_editor';
-import { HTMLAttributes, mount, ReactWrapper, shallow } from 'enzyme';
-import { CollapseSetting } from '../../shared_components/collapse_setting';
-import { EuiButtonGroup, EuiColorPicker, PropsOf } from '@elastic/eui';
-import { mountWithIntl } from '@kbn/test-jest-helpers';
-import { LayoutDirection } from '@elastic/charts';
-import { act } from 'react-dom/test-utils';
-import { EuiColorPickerOutput } from '@elastic/eui/src/components/color_picker/color_picker';
-import { createMockFramePublicAPI, generateActiveData } from '../../mocks';
-import { chartPluginMock } from '@kbn/charts-plugin/public/mocks';
-import { euiLightVars } from '@kbn/ui-theme';
-import { DebouncedInput } from '@kbn/visualization-ui-components';
 import { DatasourcePublicAPI } from '../..';
-import { CollapseFunction } from '../../../common/expressions';
-
-jest.mock('lodash', () => {
-  const original = jest.requireActual('lodash');
-
-  return {
-    ...original,
-    debounce: (fn: unknown) => fn,
-  };
-});
-
-const SELECTORS = {
-  PRIMARY_METRIC_EDITOR: '[data-test-subj="lnsMetricDimensionEditor_primary_metric"]',
-  SECONDARY_METRIC_EDITOR: '[data-test-subj="lnsMetricDimensionEditor_secondary_metric"]',
-  MAX_EDITOR: '[data-test-subj="lnsMetricDimensionEditor_maximum"]',
-  BREAKDOWN_EDITOR: '[data-test-subj="lnsMetricDimensionEditor_breakdown"]',
-};
+import { createMockFramePublicAPI, generateActiveData } from '../../mocks';
 
-const nonNumericMetricFrame = createMockFramePublicAPI({
-  activeData: generateActiveData([
-    {
-      id: 'first',
-      rows: Array(3).fill({
-        'metric-col-id': 'nonNumericData',
-        'max-col-id': 1000,
-      }),
-    },
-  ]),
-});
+const waitFor = (fn: () => void, options?: Parameters<typeof waitForRtl>[1]) =>
+  waitForRtl(fn, { timeout: 350, ...options });
```

Comment:
Why the custom timeout?

Reply:
I think I was just trying things out and forgot to remove 😅 Thanks for pointing out!